File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void rowchange(vector<vector<int>> &arr,int i){
        for(int k=0;k<arr[0].size();k++){
            if(arr[i][k]!=0)arr[i][k]=INT_MAX-10;
            
        }
    }
    void colmchange(vector<vector<int>> &arr,int j){
        for(int k=0;k<arr.size();k++){
           if(arr[k][j]!=0) arr[k][j]=INT_MAX-10;
        }
    }
    void setZeroes(vector<vector<int>>& matrix) {
        
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(matrix[i][j]==0){
                    rowchange(matrix,i);
                    colmchange(matrix,j);
                }
            }
        }
      for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(matrix[i][j]==INT_MAX-10){
                    matrix[i][j]=0;
                }
            }
        }
    
        
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    void setcolumn(int[][] matrix , int idx){
        for(int i=0;i<matrix.length;i++){
            if(matrix[i][idx]!=0) matrix[i][idx]=Integer.MAX_VALUE-10;
        }
    }
    void setrow(int[][] matrix , int idx){
        for(int i=0;i<matrix[0].length;i++){
            if(matrix[idx][i]!=0) matrix[idx][i]=Integer.MAX_VALUE-10;
        }
    }
    public void setZeroes(int[][] matrix) {
    
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]==0){
                    setrow(matrix,i);
                    setcolumn(matrix,j);
                }
            }
        }
         for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]==Integer.MAX_VALUE-10){
                    matrix[i][j]=0;
                }
            }
        }

        
    }
}